Tools/uorb_graph/create.py

```diff
@@ -259,50 +288,114 @@ def build(self, src_path_list, excluded_paths=[], use_topic_pubsub_union=True):
             fill in self._module_subsciptions & self._module_publications
         """
 
-        self._subscriptions.reset()
-        self._publications.reset()
+        self._path_blacklist = [os.path.normpath(p) for p in path_blacklist]
 
-        self._excluded_paths = [os.path.normpath(p) for p in excluded_paths]
         for path in src_path_list:
-            self._build_recursive(path)
-
-        # filter by whitelist
-        if len(self._module_whitelist) > 0:
-            self._subscriptions.filter_modules(self._module_whitelist)
-            self._publications.filter_modules(self._module_whitelist)
-
-        # modules & topics sets
-        self._modules = set(list(self._publications.pubsubs.keys()) +
-                list(self._subscriptions.pubsubs.keys()))
-        print('number of modules: '+str(len(self._modules)))
-        self._topics = self._get_topics(use_topic_pubsub_union=use_topic_pubsub_union)
-        print('number of topics: '+str(len(self._topics)))
+            log.info("## Add src path: " + path )
+            self._build_recursive(path, **kwargs)
+
+        # Summarize the found counts: (all topics are defined in 'dependency' library)
+        log.info('### Summary: Total Scanned:')
+        log.info('    Libraries Count:    '+str(len(self._found_libraries)))
+        log.info('    Modules Count:      '+str(len(self._found_modules)))
+        log.info('    Warnings Count:     '+str(len(self._warnings)))
+
+        if kwargs['merge_depends']:
+            graph.merge_depends()
+
+        # filter all scopes, topics into only the scopes + topics to output
+        self._generate_print_lists(use_topic_pubsub_union=kwargs['use_topic_pubsub_union'], merge_depends=kwargs['merge_depends'])
+
+        # Summarize the found counts:
+        print('          ### Summary (in-scope):')
```

Comment:
log.info?

Reply:
matter of taste.  

I want those lines to always show up, but `log.info` doesn`t show up by default -- it requires a `-v` cli arg.
And this isn't _really_ a warning (the next higher logging) priority level.   So I just settled for the simply option -- always output this summary.  

Of course you might argue that these summaries should be hidden, by default. 
.... I don't have a strong opinion, here.

Reply:
I don't mind much either, I just prefer to be consistent and use `log.` everywhere. That would indicate the need for another log level, but leaving it up to you.

Reply:
converted to `log.info(`